node/src/components/consensus/highway_core/state.rs

```diff
@@ -584,8 +586,20 @@ impl<C: Context> State<C> {
                 return Err(UnitError::ValueAfterTerminalBlock);
             }
         }
-        // TODO: Validate against LNC.
-        Ok(())
+        match self.validate_lnc(wunit) {
+            None => Ok(()),
+            Some(vidx) => Err(UnitError::LncNaiveCitation(vidx)),
+        }
+    }
+
+    /// Validates whether `wvote` violates the LNC rule.
+    /// Returns index of the first equivocator that should have been endorsed but wasn't.
```

Comment:
```suggestion
    /// Returns index of the first equivocator that was cited naively twice
```
…or something like that? It's not the equivocator who needs to be endorsed.

Reply:
Done.